SubTab: implementation according to GUI

TomaszLukawskiSam opened the pull request:
[Issue] https://github.com/Samsung/TAU/issues/879
[Problem] SubTab widget is not implemented
[Solution] Sub Tab widget is similar to previous TabBar.
 - added examples
 - style has been modified

Current status:
![obraz](https://user-images.githubusercontent.com/29534410/80709746-1bb64100-8aee-11ea-9576-7c1a3dc47873.png)

Test:
UIComponents > View Controls > Sub Tab



Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

TomaszLukawskiSam committed:
(mobile) SubTab: implementation according to GUI

[Issue] https://github.com/Samsung/TAU/issues/879
[Problem] SubTab widget is not implemented
[Solution]
Sub Tab widget is similar to previous TabBar.
 - widget modification and reneme
 - style has been modified
 - tests has been updated

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

TomaszLukawskiSam committed:
(mobile) SubTab: added examples

[Issue] https://github.com/Samsung/TAU/issues/879
[Problem] Lack of examples of SubTab in UIComponents
[Solution]
  - New examples for SubTab has been added
  - Older examples has been modified
  - added new method to Page widget for get height of content,
    required during Tabs widget creation

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

lslachciak reviewed (commented):
Some remarks
1) tab examples:
3 elements sub tab  has only 2 sbtabs
4 elements sub tab has only 3 subtabs
2) min width of subtab element should be 24 px (12margin left + 12 margin right)
"Tab has include 12 px margin on the left and right"
3) when I try to drag on subtab name, subtabs "jumps"
4) sutabs are not scrollable (we can discuss this more)
_"if text width is greater than the Min Width of TAB, text width increasing and tap is crolling from side to side_

TomaszLukawskiSam committed:
SubTab: fixes after review

[Issue] https://github.com/Samsung/TAU/pull/1031
[Problem] Several issues have been found in review
[Solution]
 - added left/right padding for sub tab item
 - less file for sub tab has been renamed to subtab.less

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

TomaszLukawskiSam commented:
>     2. min width of subtab element should be 24 px (12margin left + 12 margin right)
>        "Tab has include 12 px margin on the left and right"
Fixed.

TomaszLukawskiSam commented:
>     1. tab examples:
>        3 elements sub tab  has only 2 sbtabs
>        4 elements sub tab has only 3 subtabs

I don't understand this comment. Can you put here screenshot.

TomaszLukawskiSam commented:
>     3. when I try to drag on subtab name, subtabs "jumps"

I don't understand. The content of sub tab is draggable. I didn't have any problem with it.

TomaszLukawskiSam commented:
>     4. sutabs are not scrollable (we can discuss this more)
>        _"if text width is greater than the Min Width of TAB, text width increasing and tap is scrolling from side to side_

You mean about marquee effect on sub tab title? This effect is visible after fix of sub-tab padding. Current effect is `slide`. I'll change this to `endToEnd` but Marquee widget has issue and should be fixed in separate PR, I think.

lslachciak commented on examples/mobile/UIComponents/components/view-controls/sub-tab/one.html:
```diff
@@ -0,0 +1,48 @@
+					</li>
+				</ul>
+			</div>
+			<div class="ui-section-changer">
```
is this needed here?

TomaszLukawskiSam replied:
Yes. The sub-tab co-works with section changer

lslachciak commented on examples/mobile/UIComponents/components/view-controls/sub-tab/one.html:
```diff
@@ -0,0 +1,48 @@
+						<a href="#">
+							Two
+						</a>
+					</li>
```
This example has two subtabs  - similar to sub-tab-2-items.html , what is teh sense of it?

TomaszLukawskiSam replied:
You mean `two.html`, yes? This page looks the same because sample in guideline looks the same for all pages, but in real apps these file will be different.
I'm not sure what is better and clear.
Instead of this file, I would have to make change in `sub-tab-2-items.html` file. It will be strange why tab `two` links to page `one.html` 
```
<li class="ui-li-anchor">
	<a class="ui-tab-active" href="./one.html">
		One
	</a>
</li>
<li class="ui-li-anchor">
	<a href="./**one**.html">
		Two
	</a>
</li>
```
I mean `./one.html` as link for item `two`.

lslachciak commented:
> I don't understand this comment. Can you put here screenshot.

I just put comments in problematic code

lslachciak commented:
> I don't understand. The content of sub tab is draggable. I didn't have any problem with it.

This is an effect after making a drag on subtab 
![image](https://user-images.githubusercontent.com/2246331/81044342-78ca4200-8eb4-11ea-95b7-15388bed5db4.png)

TomaszLukawskiSam commented:
> > I don't understand. The content of sub tab is draggable. I didn't have any problem with it.
> 
> This is an effect after making a drag on subtab

Ok. You mean drag on sub tabs not content of subtabs.

HunseopJeong commented on src/css/profile/mobile/common/subtab.less:
```diff
@@ -0,0 +1,108 @@
+
+	a {
+		.font(regular);
+		font-size: 15 * @px_base;
```
Use @sp_base for the font-size.

TomaszLukawskiSam replied:
Done.

HunseopJeong reviewed (approved):
It looks good to me except for a comment.

TomaszLukawskiSam commented:
> > I don't understand. The content of sub tab is draggable. I didn't have any problem with it.
> 
> This is an effect after making a drag on subtab
> ![image](https://user-images.githubusercontent.com/2246331/81044342-78ca4200-8eb4-11ea-95b7-15388bed5db4.png)

Done

DohyungLim reviewed (approved):
looks good.

lslachciak commented:
I still think that there is sth wrong with navigation logic in this app. See screens below
I'm in the subtabs menu
![image](https://user-images.githubusercontent.com/2246331/81261303-2025b100-903c-11ea-8d87-525f31e3cae8.png)
**I click 3 elements link. I would expect that I can see 3 subtabs. But I see 2 subtabs and 3 tabs,**
![image](https://user-images.githubusercontent.com/2246331/81261426-60852f00-903c-11ea-800a-839f3bbb211d.png)
The same when I click 4 elements link - I see 3 subtabs, not 4
When I click 5 elements link I see 5 subtabs and this is OK

TomaszLukawskiSam commented:
> The same when I click 4 elements link - I see 3 subtabs, not 4
> When I click 5 elements link I see 5 subtabs and this is OK

Ok. Now I understand you. Fixed. You meant active main tab which was not synchronized with index.html on page before.

lslachciak reviewed (approved):
it is ok now

TomaszLukawskiSam commented:
Patch has been merged. We can close the issue.
